"""Задача №29. Решение в группах Ваня и Петя поспорили, кто быстрее решит следующую задачу: “Задана последовательность
неотрицательных целых чисел. Требуется определить значение наибольшего элемента последовательности, которая завершается первым
встретившимся нулем (число 0 не входит впоследовательность)”. Однако 2 друга оказались не
такими смышлеными. Никто из ребят не смог доконца сделать это задание. Они решили так: у кого
будет меньше ошибок в коде, тот и выиграл спор. Запомощью товарищи обратились к Вам, студентам."""

# код Вани (2 ошибки)
# n = int(input())
# max_number = 1000
# while n != 0:
#  n = int(input())
#  if max_number > n:
#   max_number = n
# print(max_number)


# код Пети (4 ошибки)
n1 = int(input())
max_number = -1
while n1 < 0:  # ошибка (меньше 0)
 n = int(input())
 if max_number < n1:
   n1 = max_number
print(n1) 

# исправленный код
n1 = int(input())
max_number = -1

while n1 >= 0:
    if n1 > max_number:
        max_number = n1
    n1 = int(input())

print(max_number)

# Правильный код
# max_value = 0  # Инициализация переменной для хранения максимального значения
# while True:
#     num = int(input("Введите неотрицательное целое число (для завершения введите 0): "))
#     if num == 0:
#         break  # Если введен ноль, прерываем цикл
    
#     if num > max_value:
#         max_value = num  # Обновляем максимальное значение, если введенное число больше предыдущего максимума

# if max_value != 0:
#     print(f"Наибольший элемент последовательности до первого встреченного нуля: {max_value}")
# else:
#     print("Последовательность не содержит чисел до нуля.")
